include/tvm/auto_scheduler/search_task.h

```diff
@@ -44,17 +44,16 @@ class HardwareParamsNode : public Object {
   int cache_line_bytes;
 
   // GPU related parameters got from device query API
-
-  /*! \brief The max shared memory per block. */
-  int max_shared_memory_per_block{INT32_MAX};
-  /*! \brief The max register memory per block. */
-  int max_registers_per_block{INT32_MAX};
-  /*! \brief The max threads per block. */
-  int max_threads_per_block{INT32_MAX};
+  /*! \brief The max shared memory per block in bytes. */
+  int max_shared_memory_per_block;
+  /*! \brief The max number of register per block. */
+  int max_registers_per_block;
+  /*! \brief The max number of threads per block. */
+  int max_threads_per_block;
```

Comment:
How about exposing them under the `tvm/_ffi/runtime_ctypes.py` like 
```python
@property
def max_thread_dimensions(self):
```

This will bring convenience when we want to get these values in our `HardwareParams` like (mali target)

Reply:
This is already supported by `VisitAttrs`

Reply:
What I meant is we add code wrapper using `attr` like this:
```python
@property
def max_registers_per_block(self):
    """Return max registers per block"""
    return self._GetDeviceAttr(self.device_type, self.device_id, 9)
```
Otherwise we access these property will not be convenient like other property `max_shared_memory_per_block`

Reply:
This is irrelevant to this PR.

Reply:
It is ok to me.

Reply:
Please take another look and approve. Thanks!